src/modules/Game.class.js

```diff
@@ -1,68 +1,165 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(
+    cells = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.cells = cells;
+    this.restart();
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  slideAndMerge(row) {
+    const filteredRow = row.filter((val) => val !== 0);
+    const newRow = [];
+
+    for (let i = 0; i < filteredRow.length; i++) {
+      if (filteredRow[i] === filteredRow[i + 1]) {
+        newRow.push(filteredRow[i] * 2);
+        this.score += filteredRow[i] * 2;
+        i++;
+      } else {
+        newRow.push(filteredRow[i]);
+      }
+    }
+
+    while (newRow.length < 4) {
+      newRow.push(0);
+    }
+
+    return newRow;
+  }
+
+  move(direction) {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    const rotated = (board) => {
+      return board[0].map((_, colIndex) => board.map((row) => row[colIndex]));
+    };
+
+    let moved = false;
+
+    if (direction === 'up' || direction === 'down') {
+      this.board = rotated(this.board);
+    }
+
+    for (let i = 0; i < 4; i++) {
+      const row =
+        direction === 'right' || direction === 'down'
+          ? [...this.board[i]].reverse()
+          : [...this.board[i]];
+
+      const newRow = this.slideAndMerge(row);
+
+      if (direction === 'right' || direction === 'down') {
+        newRow.reverse();
+      }
+
+      if (this.board[i].toString() !== newRow.toString()) {
+        this.board[i] = newRow;
+        moved = true;
+      }
+    }
+
+    if (direction === 'up' || direction === 'down') {
+      this.board = rotated(this.board);
+    }
+
+    if (moved) {
+      this.addRandomTitle();
```

Comment:
There seems to be a typo in the method name `addRandomTitle`. It should likely be `addRandomTile` to reflect the action of adding a random tile to the board.